engines/dfc_provider/spec/controllers/dfc_provider/api/products_controller_spec.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe DfcProvider::Api::ProductsController, type: :controller do
+  render_views
+
+  let(:user) { create(:user) }
+  let(:enterprise) { create(:distributor_enterprise, owner: user) }
+  let(:product) do
+    create(:simple_product, supplier: enterprise )
+  end
+  let!(:visible_inventory_item) do
+    create(:inventory_item,
+           enterprise: enterprise,
+           variant: product.variants.first,
+           visible: true)
+  end
+
+  describe('.index') do
+    context 'with authorization token' do
+      before do
+        request.env['Authorization'] = 'Bearer 123456.abcdef.123456'
+      end
+
+      context 'with an authenticated user' do
+        before do
+          allow_any_instance_of(DfcProvider::AuthorizationControl)
+            .to receive(:process)
+            .and_return(user)
+        end
+
+        context 'with an enterprise' do
+          before { get :index }
```

Comment:
There's a lot of overlapping `before` blocks in this spec. I think we normally have these `#get` calls inside the test blocks, like:

```ruby
context 'with an enterprise' do
   it 'is successful' do
      get :index
      expect(response.status).to eq 200
   end
```

Reply:
I am used to put all the test setup outside the test, but if you prefer I can put the page call inside.